crates/weaver_forge/src/lib.rs

```diff
@@ -192,13 +198,33 @@ impl TemplateEngine {
         loader: impl FileLoader + Send + Sync + 'static,
         params: Params,
     ) -> Self {
-        // Override the params defined in the `weaver.yaml` file with the params provided
-        // in the command line.
-        for (name, value) in params.params {
-            _ = config
-                .params
-                .get_or_insert_with(HashMap::new)
-                .insert(name, value);
+        // Compute the params for each template based on:
+        // - CLI-level params
+        // - Top-level params in the `weaver.yaml` file
+        if let Some(templates) = config.templates.as_mut() {
+            for template in templates {
+                let template_params = template.params.get_or_insert_with(BTreeMap::new);
+
+                // Add CLI-level params to the template params. If a param is already defined
+                // in the template params, the CLI-level param will override it.
+                // Note: The result of the insert method is ignored because we don't care about
+                // the previous value of the param.
+                for (name, value) in params.params.iter() {
+                    _ = template_params.insert(name.clone(), value.clone());
```

Comment:
Nit: can you add the comment about ignoring the previous value here?

Reply:
Done